prow/plugins/projectmanager/projectmanager.go

```diff
@@ -108,61 +115,102 @@ type githubClient interface {
 	CreateProjectCard(columnID int, projectCard github.ProjectCard) (*github.ProjectCard, error)
 }
 
-func handlePR(gc githubClient, projectManager plugins.ProjectManager, log *logrus.Entry, pe github.PullRequestEvent) error {
-	// Only respond to label add or issue/PR open events
-	if pe.Action != github.PullRequestActionOpened &&
-		pe.Action != github.PullRequestActionReopened &&
-		pe.Action != github.PullRequestActionLabeled {
+type eventData struct {
+	id     int
+	number int
+	isPR   bool
+	org    string
+	repo   string
+	state  string
+	labels []github.Label
+	remove bool
+}
+
+type DuplicateCard struct {
+	projectName string
+	issueURL    string
+}
+
+func (m *DuplicateCard) Error() string {
+	return fmt.Sprintf(issueAlreadyInProject, m.issueURL, m.projectName)
+}
+
+func handleIssueRequest(pc plugins.Agent, ie github.IssueEvent) error {
```

Comment:
I think this function is meant to be called `handleIssue`?

More importantly I think there is a misunderstanding about how issue and pull request events work. See the documentation for issue events here: https://developer.github.com/v3/issues/events/
GitHub sends Issue events for both PRs and Issues, but only sends PullRequest events to PRs. As is, your code would handle every event to a PullRequest twice. To correct this you could make the Issue event handler ignore events for Pull Requests so that they are only handled by the PR event handler, but in this case there doesn't appear to be any benefit to using PR events over Issue events. On the contrary, Issue events come with the labels, saving us an API token.
So it doesn't look like this plugin needs to handle PullRequest events at all and should instead only handle Issue events.

Note: you can check if an Issue event `ie` is for an issue or a PR by using `ie.Issue.IsPullRequest()`.

Reply:
Thanks for the review. I did misunderstood this, I agree now that Issue event is generated for all PR's and I changed the code to just handle Issue event. Please review.